applications/StructuralMechanicsApplication/custom_elements/kinematic_linear.cpp

```diff
@@ -167,10 +144,10 @@ namespace Kratos
                 }
 
                 // Operation performed: rRightHandSideVector += ExtForce*int_to_reference_weight
-                CalculateAndAdd_ExtForceContribution( row( Ncontainer, point_number ), rCurrentProcessInfo, body_force, rRightHandSideVector, int_to_reference_weight );
+                CalculateAndAdd_ExtForceContribution( this_kinematic_variables.N, rCurrentProcessInfo, body_force, rRightHandSideVector, int_to_reference_weight );
 
                 // Operation performed: rRightHandSideVector -= IntForce*int_to_reference_weight
-                noalias( rRightHandSideVector ) -= int_to_reference_weight * prod( trans( B ), stress_vector );
+                noalias( rRightHandSideVector ) -= int_to_reference_weight * prod( trans( this_kinematic_variables.B ), this_kinematic_variables.StressVector );
```

Comment:
the problem is with the tmp. I think what you do is not wrong, since you explicitly create the temporary.

however if you do
Matrix tmp = prod(D,B);
noalias(lhs) = prod(trans(B), tmp)

it is equally long and it is very clear that you create the temporary (which is correct in this context)